pkg/graphgen/lexgraphicalprod.go

```diff
@@ -0,0 +1,100 @@
+package graphgen
+
+import (
+	"fmt"
+
+	"github.com/arangodb/feed/pkg/datagen"
+)
+
+type LexicographicalProductParameters struct {
+	Left       Generatable
+	Right      Generatable
+	Prefix     string
+	StartIndex uint64
+}
+
+type edgeLabels struct {
+	from string
+	to   string
+}
+
+// We assume that Right is small and wait until we obtain the whole graph Right.
+func (u LexicographicalProductParameters) MakeGraphGenerator() (GraphGenerator, error) {
```

Comment:
We must warn in the documentation that this only works for small factors, since otherwise we run out of memory. Keep in mind that the main purpose of this tool is to put terabytes of data into a cluster quickly and then load test that. In this context, tiny graphs are normally not needed. But yes, the tool should also be able to do smaller, more interesting graphs, so this is fine, but then we need to warn users in the documentation.

Reply:
Agree. Normally, one shouldn't want to make the second graph big. The number of edges of the resulting graph is at least quadratic in the number of vertices of the second graph. There are, of course, exceptions (the complete n-partite graph).